src/Bicep.IO/Abstraction/ResourceIdentifier.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bicep.IO.Abstraction
+{
+    /// <summary>
+    /// A ResourceIdentifier is a RFC3986 URI with absolute path and without the query and fragment components.
+    /// </summary>
+    public readonly struct ResourceIdentifier : IEquatable<ResourceIdentifier>
+    {
+        public static class GlobalSettings
+        {
+            public static bool PathCaseSensitive { get; set; } = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
```

Comment:
Is this the same as the .NET implementation? AFAIK there are exceptions to this - e.g. you can choose either case-sensitive or case-insensitive on OSX.

Reply:
This is same as Omnisharp's DocumentUri implementation. `System.Uri` doesn’t implement `IEquatable`, and that's why we have custom path comparers in Bicep.Core in a few places, which is not ideal. Speaking of exceptions, technically, case-insensitive and case-senstitive file systems can exist on all OS platforms, depending on what file system the user has chosen to use. However, there’s no reliable way to detect it. Given these limitations, I believe a heuristic based on the OS platform is the most practical solution.

Reply:
You can make [_portions_ of paths case-sensitive in NTFS](https://learn.microsoft.com/en-us/windows/wsl/case-sensitivity#change-the-case-sensitivity-of-files-and-directories), too, so getting this perfect is a mug's game.

When do we need to know if a path is case sensitive or not? Presumably this is handled by the OS when we try to open the file.

Reply:
We need to address path case sensitivity when resolving module paths during module instantiation to prevent duplicate models for the same source file.